### PR TITLE
Allow passing an optional ARN when health checking  an AWSOIDC integration

### PR DESCRIPTION
**Pull request by kimlisa**
This new addition allows us to health check aws oidc integrations before a user creates or edits the integration.

Demo:

`curl --data '{"arn":"arn:aws:iam::123456789012:role/MyRole"}' '.../integrations/aws-oidc/my-integration/ping'`
```json
{
  "accountId": "123456789012",
  "arn": "arn:aws:sts::123456789012:assumed-role/MyRole/123",
  "userId": "AAA:17290"
}
```

### CODE REVIEW

**Comment by GavinFrazar** [comment]
Is the check to see if the role ARN exists (or if it's configured to trust the OIDC token) before creating the integration?

**Comment by marcoandredinis** [comment]
I'll convert it into draft because I want to polish some parts and add tests.

**Comment by marcoandredinis** [comment]
> Is the check to see if the role ARN exists (or if it's configured to trust the OIDC token) before creating the integration?

It checks if it is configured to trust the OIDC token.
So, it will fail when the role does not exist or when the IAM Role's trust relationship does not allow this particular Identity Provider's.

**Review by GavinFrazar — APPROVED** [review]

**Comment by marcoandredinis** [comment]
@fspmarshall @camscale Can you please take a look?

**Review by camscale — APPROVED** [review]
Sorry I didn't review this sooner. When I see a PR with people explicitly chosen to review it, I kind of ignore the bot assignment assuming those others have better context for the review.

File: lib/integrations/awsoidc/clients_test.go [hunk 1]
```diff
@@ -29,30 +29,30 @@ import (
 func TestCheckAndSetDefaults(t *testing.T) {
 	t.Run("invalid regions must return an error", func(t *testing.T) {
 		err := (&AWSClientRequest{
-			IntegrationName: "my-integration",
-			Token:           "token",
-			RoleARN:         "some-arn",
-			Region:          "?",
+			//IntegrationName: "my-integration",
```
[the remark this answers is not in the document]
Comment (marcoandredinis) [reply lib/integrations/awsoidc/clients_test.go:32]: Removed, thank you.

**Comment by marcoandredinis** [comment]
@kimlisa After changing some things, would you mind taking a look?
Given that you created the PR, you can't actually approve it but let me know if it looks good and I'll ask for a G1 bot review.

**Comment by kimlisa** [comment]
approved

**Review by r0mant — APPROVED** [review]
Bot.

**Pull request merged**